### DISCUSSION TITLE
Default params of avif conversion.

### DISCUSSION THREAD

**Discussion by dAvagimyan** [libvips/libvips#3937] [Help wanted]
I have a some issue in  [govips](https://github.com/davidbyttow/govips/issues/423).  The general meaning of the appeal there is that my conversion to avif through the code produces worse compression than through a vips copy in console. There is no such behavior with webp conversion

size -> 1327450 -> cats.jpg - original
size ->1003222 -> cats.avif - via go code
size -> 404594 -> cats_inconsole.avif - via console (vips copy cats.jpg cats.avif )

hey suggested that the problem may be that govips probably does not have all the parameters. Is it possible to find out the default parameters when converting jpeg to avif?   To be honest, I don’t know where to dig next

```
vips version:
vips-8.15.1

 heif-enc --list-encoders
HEIC encoders:
AVIF encoders:
- aom = AOMedia Project AV1 Encoder 3.8.2-377-g04d6253 [default]
- svt = SVT-AV1 encoder v1.8.0
JPEG encoders:
- jpeg = libjpeg-turbo 1.2.90 (libjpeg 6.2) [default]
JPEG 2000 encoders:
Uncompressed encoders:
- uncompressed = uncompressed [default]
```

**Comment by lovell** [comment]
Did you see https://www.libvips.org/API/current/VipsForeignSave.html#vips-heifsave ?

**Reply by dAvagimyan** [reply]
Nope. Thanks a lot! As i understand avif compression is not better webp in equal quality (hight)?

**Reply by jcupitt** [reply]
It depends on the image and your needs, of course, but for lossy compression of photographs, avif, heic and jxl all make files that are about 30 - 50% smaller than webp at the same quality. webp and mozjpeg are about the same, then original jpeg is about 30% larger again.

webp is a useful catch-all format, but doesn't have a really compelling compression algorithm. That's why google mostly abandoned development 10 years ago.

https://cloudinary.com/blog/jpeg-xl-and-the-pareto-front

Is mostly about jxl, but they have some nice graphs which include avif, webp, mozjpeg and original jpeg, for example:

https://res.cloudinary.com/cloudinary-marketing/images/f_auto,q_auto/v1709058526/Web_Assets/blog/blog-pareto-front-5/blog-pareto-front-5-png

That's at the same quality with bits per pixel on the horizontal axis. So original jpeg is 0.83, mozjpeg and webp are about 0.7, avif and jxl (and heic) are 0.55.

**Reply by dAvagimyan** [reply]
Thank you, now I understand
